fairscale/utils/testing.py

```diff
@@ -159,19 +160,19 @@ def worker_process(rank: int, world_size: int, filename: str, func: Callable, ar
     initialize_model_parallel(1, world_size, **kwargs)
     try:
         func(*args)
+        teardown()
     except BaseException as e:
+        # Make sure that the group is properly destroyed, even for tests which check for exceptions being raised
+        teardown()
```

Comment:
not entirely sure, but I'm guessing that it could be why we were seeing issues, if that exception was raised then the teardown was skipped